polyfill/lib/date.mjs

```diff
@@ -120,11 +120,12 @@ export class Date {
       calendar = GetSlot(this, CALENDAR);
       source = this;
     }
-    const props = ES.ToPartialRecord(temporalDateLike, ['day', 'era', 'month', 'year']);
+    const fieldNames = calendar.fields(['day', 'era', 'month', 'year']);
+    const props = ES.ToPartialRecord(temporalDateLike, fieldNames);
```

Comment:
This is too trusting of the calendar - should make sure we've got a clean array of strings. (Potentially for followup.)

Reply:
I think fixing the next comment below (https://github.com/tc39/proposal-temporal/pull/1014#discussion_r509278096) will be less work if I also fix this, so I won't defer it to a followup. To be clear, I assume you are suggesting moving this into an abstract operation named something like CalendarFields, which does some sort of validation on the returned value?